### PR TITLE
feature/registrator - interfaces refactor

### PR DESCRIPTION
**Pull request by JozefFlakus**
**Problem:**
It was hard to find proper interface definition in registrator module structure.

**Solution:**
- removed redundant `ConfigState` interface = simplified config types definition
- separated and moved interfaces to dedicated folder and files
- moved mongoose related files to separated `mongoose` folder - some cleanup for future PostgreSQL adapters

### CODE REVIEW

File: src/registrator/registrator.module.ts [hunk 1]
```diff
@@ -1,39 +1,35 @@
...
+  const { databaseAdapter, databaseUri, app } = config;
 
-  if (!DB_URI) {
-    return handleError('DB_URI not provided in config');
-  }
-
-  if (!EXPRESS_APP) {
-    return handleError('EXPRESS_APP not provided in config');
-  }
+  if (!app) { return handleError('EXPRESS_APP not provided in config'); }
```
Comment (pdomaleczny) [inline src/registrator/registrator.module.ts]: I guess we don't need the brackets here :D

File: src/registrator/registrator.module.ts [hunk 2]
```diff
@@ -1,39 +1,35 @@
...
+  if (!databaseUri) { return handleError('DB_URI not provided in config'); }
```
Comment (pdomaleczny) [inline src/registrator/registrator.module.ts]: Here also ☝️

File: src/registrator/registrator.module.ts [hunk 3]
```diff
@@ -1,39 +1,35 @@
...
 
-  switch (DB_ADAPTER) {
-    case 'mongo':
+  switch (databaseAdapter) {
+    case 'mongoose':
       return useAdapter(mongooseAdapter)({ name, schema, config });
   }
+
+  return handleError('Not recognized database adapter was used');
```
Comment (pdomaleczny) [inline src/registrator/registrator.module.ts]: do on `default:`?

File: src/registrator/registrator.module.ts (hunk 1 above)
Comment (JozefFlakus) [reply src/registrator/registrator.module.ts]: we have `tslint` configured to warn `if` statements without brackets

File: src/registrator/registrator.module.ts (hunk 2 above)
Comment (JozefFlakus) [reply src/registrator/registrator.module.ts]: we have `tslint` configured to warn `if` statements without brackets

**Pull request merged**